servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -6726,6 +6726,49 @@ public void putTenancy(ResourceContext ctx, String tenantDomain, String provider
         if (authzServiceTokenOperation) {
             setupTenantAdminPolicyInProvider(ctx, provSvcDomain, provSvcName, tenantDomain,
                     false, auditRef, caller);
+
+            // If listed as service provider, register dependency between the provider service and the tenant domain (unless dependency exist
+            // following creation of a previous resource group)
+
+            tenancyRegisterDomainDependency(ctx, tenantDomain, provider, true, auditRef, caller);
+        }
+    }
+
+    private void tenancyRegisterDomainDependency(ResourceContext ctx, String tenantDomain, String provider, boolean authzServiceTokenOperation, String auditRef, String caller) {
+        String serviceToRegister = provider;
+        try {
+            serviceToRegister = getAuthorizedProviderService((RsrcCtxWrapper) ctx, serviceToRegister, authzServiceTokenOperation, caller);
```

Comment:
I'm not sure if I understand why are we making this call here? It looks to be in all cases we're dealing with the provider value that is passed in explicitly in the url or the body of the object. So why are we trying to possible change it - what is the use case where the provider value is not correct and we need to change it?

Reply:
For example, the putTenant call can be made by a service provider without an authorized service token. In this case, wouldn't we want to use the principal as the dependent service provider instead of the service specified in the body? or would we like to use the domain and service the principal passed? (as long as it is authorized to do the "putTenant" call for that specific domain and service)

Reply:
Well the putTenant call can be made by a principal which is different than the service provider it passed in the request. But I see now that even if that is the case, we would still want to use the provider it passed in the request and not itself as the dependent service (as long as the principal is allowed to call putTenant with a specific provider service, we can list that provider as the dependent)